### PR TITLE
Eng 1614 add update timestamps for workflow

### PR DESCRIPTION
**Pull request by likawind**
## Describe your changes and why you are making these changes
This PR adds `execState` to the entire workflow execution so that we can also track the orchestration's exec status, timestamps, and error logs. It unblocks https://linear.app/aqueducthq/issue/ENG-1613/give-user-more-insight-into-a-system-internal-dag-error to add more error tracking to workflow when failures are outside operators.

The new field is visible in the new api `workflow/{wfId}/result/{dagResultId}`.

## Related issue number (if any)
ENG-1614 ENG-1613

## Checklist before requesting a review
Manually verified migration. Others are covered by integration tests.
Manually verified response:
```
"dag_id": "1420193b-b7b5-4e1d-b725-f9413b536b78",
    "dag_created_at": "2022-08-09T05:06:23.348466009Z",
    "storage_config": {
        "type": "file",
        "file_config": {
            "directory": "/home/ubuntu/.aqueduct/server/storage"
        }
    },
    "engine_config": {
        "type": "aqueduct",
        "aqueduct_config": {}
    },
    "workflow_id": "ee9bf39a-f1bd-47d1-a9f6-3448ef747eab",
    "workflow_created_at": "2022-08-09T05:01:48.788283751Z",
    "user_id": "0678fdc2-1641-4d42-98e4-8e4566825ba1",
    "name": "Demo Churn Ensemble 1",
    "description": "",
    "schedule": {
        "trigger": "periodic",
        "cron_schedule": "0 * * * *",
        "disable_manual_trigger": false,
        "paused": false
    },
    "retention_policy": {
        "k_latest_runs": -1
    },
    "result": {
        "id": "460abc2e-4841-416d-94b5-1d99e5abbc0f",
        "exec_state": {
            "user_logs": null,
            "status": "succeeded",
            "failure_type": null,
            "error": null,
            "timestamps": {
                "registered_at": null,
                "pending_at": "2022-09-13T19:00:00.023309323Z",
                "running_at": "2022-09-13T19:00:00.026183872Z",
                "finished_at": "2022-09-13T19:00:20.074535868Z"
            }
        }
    },
```
- [x] I have created a descriptive PR title. The PR title should complete the sentence "This PR...".
- [x] I have performed a self-review of my code.
- [x] I have included a small demo of the changes. For the UI, this would be a screenshot or a Loom video.
- [ ] If this is a new feature, I have added unit tests and integration tests.
- [x] I have run the integration tests locally and they are passing.
- [x] I have run the linter script locally (See `python3 scripts/run_linters.py -h` for usage).
- [x] All features on the UI continue to work correctly.
- [x] Added one of the following CI labels:
    - `run_integration_test`: Runs integration tests
    - `skip_integration_test`: Skips integration tests (Should be used when changes are ONLY documentation/UI)

### CODE REVIEW

File: src/golang/lib/airflow/writer.go [hunk 1]
```diff
@@ -37,23 +31,17 @@ func createWorkflowDagResult(
 		return nil, errors.New("Cannot create WorkflowDagResult for in progress Airflow DAG Run.")
 	}
 
-	workflowDagResult, err := workflowDagResultWriter.CreateWorkflowDagResult(ctx, dbDag.Id, db)
-	if err != nil {
-		return nil, err
-	}
-
-	changes := map[string]interface{}{
-		workflow_dag_result.CreatedAtColumn: *run.StartDate.Get(),
```
Comment (saurav-c) [inline src/golang/lib/airflow/writer.go:46]: We need the `CreatedAt` column to be in order to perform proper syncing. Se https://github.com/aqueducthq/aqueduct/blob/main/src/golang/lib/airflow/sync.go#L133

For now can you add an `updateWorkflowDagResult` call that sets the `CreatedAt` column.

File: src/golang/lib/collections/tests/workflow_dag_result_test.go [hunk 2]
```diff
@@ -12,15 +13,25 @@ import (
 
 // seedWorkflowDagResultWithDags populates the workflow_dag_result table with count
 // workflow dag results where workflow_dag_id is set to the values provided in workflowDagIds.
-func seedWorkflowDagResultWithDags(t *testing.T, count int, workflowDagIds []uuid.UUID) []workflow_dag_result.WorkflowDagResult {
+func seedWorkflowDagResultWithDags(
+	t *testing.T,
+	count int,
+	now time.Time,
```
Comment (saurav-c) [inline src/golang/lib/collections/tests/workflow_dag_result_test.go:19]: Why does the timestamp need to be passed in here? The `seed` call should be used when you just need to create `workflowDagResults` as easily as possible. Lets remove any arguments that aren't needed.

For testing a specific columns value we should explicitly create the record.

File: src/golang/lib/collections/workflow_dag_result/sqlite.go [hunk 3]
```diff
@@ -51,17 +50,22 @@ func (r *sqliteReaderImpl) GetKOffsetWorkflowDagResultsByWorkflowId(
...
+	insertColumns := []string{IdColumn, WorkflowDagIdColumn, StatusColumn, CreatedAtColumn, ExecStateColumn}
 	insertWorkflowDagResultStmt := db.PrepareInsertWithReturnAllStmt(tableName, insertColumns, allColumns())
 
 	id, err := utils.GenerateUniqueUUID(ctx, tableName, db)
 	if err != nil {
 		return nil, err
 	}
 
-	args := []interface{}{id, workflowDagId, shared.PendingExecutionStatus, time.Now()}
+	if execState.Timestamps == nil || execState.Timestamps.PendingAt == nil {
```
Comment (saurav-c) [inline src/golang/lib/collections/workflow_dag_result/sqlite.go:64]: This should be commented somewhere that the pendingat timestamp can't be nil.

File: src/golang/lib/collections/workflow_dag_result/workflow_dag_result.go [hunk 4]
```diff
@@ -9,14 +9,18 @@ import (
...
+
 type WorkflowDagResult struct {
-	Id            uuid.UUID              `db:"id" json:"id"`
-	WorkflowDagId uuid.UUID              `db:"workflow_dag_id" json:"workflow_dag_id"`
-	Status        shared.ExecutionStatus `db:"status" json:"status"`
-	CreatedAt     time.Time              `db:"created_at" json:"created_at"`
+	Id            uuid.UUID                 `db:"id" json:"id"`
+	WorkflowDagId uuid.UUID                 `db:"workflow_dag_id" json:"workflow_dag_id"`
+	Status        shared.ExecutionStatus    `db:"status" json:"status"`
+	CreatedAt     time.Time                 `db:"created_at" json:"created_at"`
```
Comment (saurav-c) [inline src/golang/lib/collections/workflow_dag_result/workflow_dag_result.go:22]: Are you planning to deprecate `CreatedAt`?

**Review by saurav-c — CHANGES_REQUESTED** [review]

File: src/golang/lib/collections/workflow_dag_result/workflow_dag_result.go (hunk 4 above)
Comment (likawind) [reply src/golang/lib/collections/workflow_dag_result/workflow_dag_result.go:22]: Yes, there are too many APIs (mainly in frontend) using that field so we would need to do that after we clean up these APIs

File: src/golang/lib/airflow/writer.go (hunk 1 above)
Comment (likawind) [reply src/golang/lib/airflow/writer.go:46]: idk why this comment is missing in file view...I believe we set this field while creating the DAG result using the execState. Am I missing anything?

File: src/golang/lib/collections/workflow_dag_result/sqlite.go (hunk 3 above)
Comment (likawind) [reply src/golang/lib/collections/workflow_dag_result/sqlite.go:64]: Added a few comments in the interface struct. Also created https://linear.app/aqueducthq/issue/ENG-1701/deprecate-createdat-field-for-workflow-dag-results to remove this constraints once that is done.

File: src/golang/lib/collections/workflow_dag_result/workflow_dag_result.go (hunk 4 above)
Comment (likawind) [reply src/golang/lib/collections/workflow_dag_result/workflow_dag_result.go]: https://linear.app/aqueducthq/issue/ENG-1701/deprecate-createdat-field-for-workflow-dag-results

**Review by saurav-c — APPROVED** [review]
Changes look good!

**Pull request merged**